src/orion/core/worker/experiment.py

```diff
@@ -219,13 +219,16 @@ def register_trials(self, trials):
 
         :type trials: list of `Trial`
         """
-        stamp = datetime.datetime.utcnow()
-        for trial in trials:
-            trial.experiment = self._id
-            trial.status = 'new'
-            trial.submit_time = stamp
-        trials_dicts = list(map(lambda x: x.to_dict(), trials))
-        self._db.write('trials', trials_dicts)
+        try:
+            stamp = datetime.datetime.utcnow()
+            for trial in trials:
+                trial.experiment = self._id
+                trial.status = 'new'
+                trial.submit_time = stamp
+            trials_dicts = list(map(lambda x: x.to_dict(), trials))
+            self._db.write('trials', trials_dicts)
+        except DuplicateKeyError:
+            pass
```

Comment:
Could it be the case that we want this to be reraised? If determinism in the sequence of generated trials was ensured by a future PR, the behaviour caught here would be unacceptable, and it should not have been silenced with a `pass` clause.

Reply:
The plan would be to either let the error climb or catch and return True/False to confirm if the trial was registered properly. Also, if we expect writes on db to raise duplicatekey errors, it may be better to turn this function into `register_trial()` instead of `register_trials()`.

Reply:
@tsirif @DatCorno I was thinking about this and the particular issue here is really related to non-blocking algo sampling design. I propose we leave this methods as-is in this PR, maybe with a clear TODO comment like "This method should be refactored to accept one trial at a time only, and return true or false if trial was registered properly or not". The PR fixing this should be done before end of this week.

Reply:
I am not sure why this error needs to be caught and get silenced.

I argue that this behaviour, if it ever appears in the future design, it indicates that determinism was not implemented correctly. So this would be our software's error, and we shouldn't let it pass silently because we will never be aware of it. Even if log an error message, if the event is rare enough, this is not a reliable way to have it issued. Having said these, I believe:
1. that we should use the error reporting system of Python here, which is raising exceptions
2. not silence the exception, but at least let the native one break the program, or reraising it from another orion-specific exception to include extra semantical information about how it happened.

I am in favour of making `register_trials` -> `register_trial` and iterating over them in `Producer`.

Reply:
Could we do this now?

Reply:
The plan is to silence it only in this PR and fix this in the PR about non-blocking design for algo sampling. When we have this latter PR, this kind of error can be catched in the producer and considered a race condition, meaning that the algo should be updated again before resampling.

Reply:
Doing it now would mean implementing the non-blocking parallel sampling in this PR.